Mage.Sets/src/mage/cards/v/VeteranBodyguard.java

```diff
@@ -0,0 +1,132 @@
+ *  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ *  ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ *  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ *  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *  The views and conclusions contained in the software and documentation are those of the
+ *  authors and should not be interpreted as representing official policies, either expressed
+ *  or implied, of BetaSteward_at_googlemail.com.
+ */
+package mage.cards.v;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.condition.InvertCondition;
+import mage.abilities.condition.common.SourceTappedCondition;
+import mage.abilities.decorator.ConditionalContinuousEffect;
+import mage.abilities.effects.ReplacementEffectImpl;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.constants.Zone;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.permanent.UnblockedPredicate;
+import mage.game.Game;
+import mage.game.events.DamageEvent;
+import mage.game.events.DamagePlayerEvent;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+
+/**
+ *
+ * @author MTGfan
+ */
+public class VeteranBodyguard extends CardImpl {
+
+    public VeteranBodyguard(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{3}{W}{W}");
+        
+        this.subtype.add("Human");
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(5);
+
+        // As long as Veteran Bodyguard is untapped, all damage that would be dealt to you by unblocked creatures is dealt to Veteran Bodyguard instead.
+        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new ConditionalContinuousEffect(new VeteranBodyguardEffect(), new InvertCondition(new SourceTappedCondition()), "As long as {this} is untapped, all damage that would be dealt to you by unblocked creatures is dealt to {this} instead.")));
+    }
+
+    public VeteranBodyguard(final VeteranBodyguard card) {
+        super(card);
+    }
+
+    @Override
+    public VeteranBodyguard copy() {
+        return new VeteranBodyguard(this);
+    }
+}
+
+class VeteranBodyguardEffect extends ReplacementEffectImpl {
+
+    private static final FilterCreaturePermanent filter = new FilterCreaturePermanent("unblocked creatures");
+    
+    static {
+        filter.add(new UnblockedPredicate());
+    }
+    
+    VeteranBodyguardEffect() {
+        super(Duration.EndOfTurn, Outcome.RedirectDamage);
+        staticText = "all combat damage that would be dealt to you by unblocked creatures is dealt to {source} instead.";
```

Comment:
Digging around in the cards already implemented I see Martyrs Of Korlis which is very similar with: "As long as Martyrs of Korlis is untapped, all damage that would be dealt to you by artifacts is dealt to Martyrs of Korlis instead." Perhaps somehow between these two is the correct way to code Veteran Bodyguard? And the implementation of Martyrs of Koralis looks cleaner see below:

```
        this.addAbility(new SimpleStaticAbility(Zone.BATTLEFIELD, new ConditionalContinuousEffect(
                new RedirectArtifactDamageFromPlayerToSourceEffect(Duration.WhileOnBattlefield),
                new InvertCondition(new SourceTappedCondition()),
                "{this} redirects artifact damage from controller as long as it's untapped")));
```

and

```
class RedirectArtifactDamageFromPlayerToSourceEffect extends RedirectionEffect {

    public RedirectArtifactDamageFromPlayerToSourceEffect(Duration duration) {
        super(duration);        
    }

    public RedirectArtifactDamageFromPlayerToSourceEffect(final RedirectArtifactDamageFromPlayerToSourceEffect effect) {
        super(effect);
    }

    @Override
    public RedirectArtifactDamageFromPlayerToSourceEffect copy() {
        return new RedirectArtifactDamageFromPlayerToSourceEffect(this);
    }

    @Override
    public boolean applies(GameEvent event, Ability source, Game game) {
        if (event.getTargetId().equals(source.getControllerId())&&
                game.getPermanentOrLKIBattlefield(event.getSourceId()).getCardType().contains(CardType.ARTIFACT)) {
            this.redirectTarget.updateTarget(source.getSourceId(), game);
            return true;
        }
        return false;
    }
}
```